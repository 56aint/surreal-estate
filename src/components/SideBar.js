import React, { useState } from "react";
// import styled from "styled-components";
import { Link, useLocation, useHistory } from "react-router-dom";
import qs from "qs";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

import "../styles/SideBar.css";

const SideBar = () => {
  const { search } = useLocation();

  const buildQueryString = (operation, valueObj) => {
    const currentQueryParams = qs.parse(search, { ignoreQueryPrefix: true });

    const newQueryParams = {
      ...currentQueryParams,
      [operation]: JSON.stringify({
        ...JSON.parse(currentQueryParams[operation] || "{}"),
        ...valueObj,
      }),
    };
    return qs.stringify(newQueryParams, {
      addQueryPrefix: true,
      encode: false,
    });
  };

  const [query, setQuery] = useState("");

  const history = useHistory();

  const handleSearch = (event) => {
    event.preventDefault();
    const newQueryString = buildQueryString("query", {
      title: { $regex: query },
    });
    console.log(query);
    history.push(newQueryString);
  };

  const handleInputChange = (e) => {
    setQuery(e.target.value);
  };

  return (
    <nav role="navigation">
      <form
        className="property-search-form"
        onChange={(e) => handleInputChange(e)}
        onSubmit={(e) => handleSearch(e)}
      >
        <input
          type="text"
          value={query}
          onChange={(e) => setQuery(e.target.value)}
        />

        <button type="submit" className="property-search-button">
          <FontAwesomeIcon icon={faSearch} />
        </button>
      </form>
      <p>View</p>
      <p>Options</p>
      <div className="sidebarToggle">
        <input name="toggle" type="checkbox" />
        <label htmlFor="toggle">
          <span>menu</span>
          <div />
          <div />
          <div />
        </label>

        <ul className="sidebar">
          <p className="sidebar-title">Filter by City</p>
          <Link to={buildQueryString("query", { city: "Manchester" })}>
            <li>Manchester</li>
          </Link>
          <Link to={buildQueryString("query", { city: "Leeds" })}>
            <li>Leeds</li>
          </Link>
          <Link to={buildQueryString("query", { city: "Shefield" })}>
            <li>Shefield</li>
          </Link>
          <Link to={buildQueryString("query", { city: "Liverpool" })}>
            <li>Liverpool</li>
          </Link>
          <p className="sidebar-title">Sort by</p>
          <Link to={buildQueryString("sort", { price: 1 })}>
            <li>Price Ascending</li>
          </Link>
          <Link to={buildQueryString("sort", { price: -1 })}>
            <li>Price Descending</li>
          </Link>
        </ul>
      </div>
    </nav>
  );
};

export default SideBar;
