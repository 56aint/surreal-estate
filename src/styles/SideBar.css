p{
  font-weight: bolder;
  display: block;
  text-align: left;
  margin-top: 0em ;
  margin-bottom: 0em ;
  /*line-height: 1.5;*/
  text-indent: 10px ;
  color: red;
}
.sidebarToggle {
  /*display: block;*/
   /*height: 100%;
   width: 160px;*/
   position: relative;
   display: flex;
   flex-direction: column;
   margin-left: 0px;
   padding: 10px;
 
   top: 0px;
   left: 0px;
   z-index: 1;
   -webkit-user-select: none;
   -moz-user-select: none;
    -ms-user-select: none;
        user-select: none;
 }
 
 .sidebarToggle a {
  display: block;
  text-decoration: none;
  color: rgb(233, 214, 214);
  -webkit-transition: color 0.3s ease;
  -o-transition: color 0.3s ease;
  transition: color 0.3s ease;
  padding: 6px 8px 6px 16px;
  background: rgb(153, 16, 16);
  width: fit-content;
  height: fit-content;
}

.sidebarToggle a:hover {
  color: #0b66a7;
}


/* The hamburger button hidden inside an input element */
.sidebarToggle input {
  display: block;
  width: 30px;
  height: 30px;
  position: absolute;
  top: -7px;
  left: -5px;
  padding: 0;
  cursor: pointer;
  opacity: 0; /* hide this */
  z-index: 2; /* and place it over the hamburger */
  -webkit-touch-callout: none;
}

.property-search-form{
  width: 25%;
  height: 50px;
}
.property-search-button{
  width: 25%;
  height: 32px;
  
  /*top: 100%;*/
  transform: translate(5%, 5%);
}

.sidebarToggle label {
  position: relative;
  display: block;
  width: 30px;
  height: 30px;
}
.sidebarToggle label span {
  display: none;
}

/*
 * Hamburger
 */
.sidebarToggle label div {
  display: block;
  width: 30px;
  height: 2px;
  margin-bottom: 6px;
  position: relative;
  background: #90da06;
  border-radius: 3px;
  z-index: 1;
  -webkit-transform-origin: 4px 0px;
      -ms-transform-origin: 4px 0px;
          transform-origin: 4px 0px;
  -webkit-transition: all 0.3s ease;
  -o-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.sidebarToggle label div:first-child {
  -webkit-transform-origin: 0% 100%;
      -ms-transform-origin: 0% 100%;
          transform-origin: 0% 100%;
}

.sidebarToggle label div:nth-child(2) {
    -webkit-transform-origin: 0% 0%;
      -ms-transform-origin: 0% 0%;
          transform-origin: 0% 0%;
}

/*
 * Change the color of the hamburger when toggled
 */
.sidebarToggle input:checked + label div {
  background: #a71313;
}

/*
 * Use absolute positioning to place the menu off-screen
 */
.sidebar {
  position: absolute;
  width: 200px;
  margin: -100px 0 0 -50px;
  padding: 50px;
  padding-top: 125px;
  background: #ffffff00;
  list-style-type: none;
  /* Hide the menu off-screen to the left */
  left: -100%;
  -webkit-transition: left 0.3s ease;
  -o-transition: left 0.3s ease;
  transition: left 0.3s ease;
}

.sidebar li {
  padding: 10px 2px;
  font-size: 12px;
}

/*
 * Slide the menu in from left when the input is checked
 */
.sidebarToggle input:checked ~ ul {
  left: 0;
}

/*loading*/
.loading {
  margin: 100px auto;
  width: 40px;
  height: 40px;
  position: relative;
  text-align: center;

  -webkit-animation: sk-rotate 2s infinite linear;
  animation: sk-rotate 2s infinite linear;
}
.spinner1,
.spinner2 {
  width: 60%;
  height: 60%;
  display: inline-block;
  position: absolute;
  top: 0;
  background-color: rgb(184, 36, 36);
  border-radius: 100%;

  -webkit-animation: sk-bounce 2s infinite ease-in-out;
  animation: sk-bounce 2s infinite ease-in-out;
}

.spinner2 {
  top: auto;
  bottom: 0;
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

@-webkit-keyframes sk-rotate {
  100% {
    -webkit-transform: rotate(360deg);
  }
}
@keyframes sk-rotate {
  100% {
    transform: rotate(360deg);
    -webkit-transform: rotate(360deg);
  }
}
@-webkit-keyframes sk-bounce {
  0%,
  100% {
    -webkit-transform: scale(0);
  }
  75% {
    -webkit-transform: scale(1);
  }
}

@keyframes sk-bounce {
  0%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  75% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}