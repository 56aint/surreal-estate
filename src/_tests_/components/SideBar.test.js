import React from "react";
import { render } from "@testing-library/react";
import SideBar from "../../components/SideBar";

describe("Sidebar", () => {
  const { asFragment } = render(<SideBar />);

  xit("render correctly", () => {
    
  });
});
