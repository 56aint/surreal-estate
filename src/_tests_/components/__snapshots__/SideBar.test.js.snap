// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correct props 1`] = `
<DocumentFragment>
  <nav
    role="navigation"
  >
    <form
      class="property-search-form"
    >
      <input
        type="text"
        value=""
      />
      <button
        class="property-search-button"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-search fa-w-16 "
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
      </button>
    </form>
    <p>
      View
    </p>
    <p>
      Options
    </p>
    <div
      class="sidebarToggle"
    >
      <input
        name="toggle"
        type="checkbox"
      />
      <label
        for="toggle"
      >
        <span>
          menu
        </span>
        <div />
        <div />
        <div />
      </label>
      <ul
        class="sidebar"
        data-testid="sidebar-id"
      >
        <p
          class="sidebar-title"
        >
          Filter by
        </p>
        <p>
           City
        </p>
        <a
          href="/?query={\\"city\\":\\"Manchester\\"}"
        >
          <li>
            Manchester
          </li>
        </a>
        <a
          href="/?query={\\"city\\":\\"Leeds\\"}"
        >
          <li>
            Leeds
          </li>
        </a>
        <a
          href="/?query={\\"city\\":\\"Shefield\\"}"
        >
          <li>
            Shefield
          </li>
        </a>
        <a
          href="/?query={\\"city\\":\\"Liverpool\\"}"
        >
          <li>
            Liverpool
          </li>
        </a>
        <p
          class="sidebar-title"
        >
          Sort by
        </p>
        <a
          href="/?sort={\\"price\\":1}"
        >
          <li>
            Price Ascending
          </li>
        </a>
        <a
          href="/?sort={\\"price\\":-1}"
        >
          <li>
            Price Descending
          </li>
        </a>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
